### ISSUE TITLE
Disappearing cursor in diff view on whitespace on added lines

### ISSUE THREAD

**Issue by ibutra** [sainnhe/everforest#72]
### Operating system/version

Windows 10

### GUI client/version

GVim 8.2

### Feature matrix

```
VIM - Vi IMproved 8.2 (2019 Dec 12 kompiliert am Apr 21 2022 22:02:12)
MS-Windows 64 Bit GUI Version mit OLE-Unterstützung
Inklusive der Patches: 1-4803
Übersetzt von appveyor@APPVYR-WIN
Riesige Version mit GUI. Ein- (+) oder ausschließlich (-) der Eigenschaften:
+acl                +ex_extra           +mzscheme/dyn       -tcl
+arabic             +extra_search       +netbeans_intg      -termguicolors
+autocmd            -farsi              +num64              +terminal
+autochdir          +file_in_path       +ole                -termresponse
+autoservername     +find_in_path       +packages           +textobjects
+balloon_eval       +float              +path_extra         +textprop
-balloon_eval_term  +folding            +perl/dyn           -tgetent
+browse             -footer             +persistent_undo    +timers
++builtin_terms     +gettext/dyn        +popupwin           +title
+byte_offset        -hangul_input       -postscript         +toolbar
+channel            +iconv/dyn          +printer            +user_commands
+cindent            +insert_expand      +profile            +vartabs
+clientserver       +ipv6               +python/dyn         +vertsplit
+clipboard          +job                +python3/dyn        +vim9script
+cmdline_compl      +jumplist           +quickfix           +viminfo
+cmdline_hist       +keymap             +reltime            +virtualedit
+cmdline_info       +lambda             +rightleft          +visual
+comments           +langmap            +ruby/dyn           +visualextra
+conceal            +libcall            +scrollbind         +vreplace
+cryptv             +linebreak          +signs              -vtp
+cscope             +lispindent         +smartindent        +wildignore
+cursorbind         +listcmds           +sodium/dyn         +wildmenu
+cursorshape        +localmap           +sound              +windows
+dialog_con_gui     +lua/dyn            +spell              +writebackup
+diff               +menu               +startuptime        -xfontset
+digraphs           +mksession          +statusline         -xim
+directx            +modify_fname       -sun_workshop       +xpm_w32
-dnd                +mouse              +syntax             -xterm_save
-ebcdic             +mouseshape         +tag_binary         
+emacs_tags         +multi_byte_ime/dyn -tag_old_static     
+eval               +multi_lang         -tag_any_white      
          System-vimrc-Datei: "$VIM\vimrc"
        Benutzer-vimrc-Datei: "$HOME\_vimrc"
 zweite Benutzer-vimrc-Datei: "$HOME\vimfiles\vimrc"
 dritte Benutzer-vimrc-Datei: "$VIM\_vimrc"
         Benutzer-exrc-Datei: "$HOME\_exrc"
 zweite Benutzer-exrc-Datei: "$VIM\_exrc"
         System-gvimrc-Datei: "$VIM\gvimrc"
       Benutzer-gvimrc-Datei: "$HOME\_gvimrc"
zweite Benutzer-gvimrc-Datei: "$HOME\vimfiles\gvimrc"
dritte Benutzer-gvimrc-Datei: "$VIM\_gvimrc"
      defaults Datei: "$VIMRUNTIME\defaults.vim"
           System-Menü-Datei: "$VIMRUNTIME\menu.vim"
Übersetzt: cl -c /W3 /GF /nologo -I. -Iproto -DHAVE_PATHDEF -DWIN32  -DFEAT_CSCOPE -DFEAT_TERMINAL 
-DFEAT_SOUND -DFEAT_NETBEANS_INTG -DFEAT_JOB_CHANNEL -DFEAT_IPV6   -DFEAT_XPM_W32 
-DHAVE_SODIUM -DDYNAMIC_SODIUM -DDYNAMIC_SODIUM_DLL=\"libsodium.dll\" /I "C:\libsodium\include" 
-DWINVER=0x0501 -D_WIN32_WINNT=0x0501 /source-charset:utf-8 /MP -DHAVE_STDINT_H /Ox /GL -DNDEBUG 
/Zl /MT /D_CRT_SECURE_NO_DEPRECATE /D_CRT_NONSTDC_NO_DEPRECATE -DFEAT_OLE -DFEAT_MBYTE_IME 
-DDYNAMIC_IME -DFEAT_GUI_MSWIN -DFEAT_DIRECTX -DDYNAMIC_DIRECTX -DFEAT_DIRECTX_COLOR_EMOJI 
-DDYNAMIC_ICONV -DDYNAMIC_GETTEXT -DFEAT_LUA -DDYNAMIC_LUA -DDYNAMIC_LUA_DLL=\"lua54.dll\" 
-DFEAT_PYTHON -DDYNAMIC_PYTHON -DDYNAMIC_PYTHON_DLL=\"python27.dll\" -DFEAT_PYTHON3 
-DDYNAMIC_PYTHON3 -DDYNAMIC_PYTHON3_DLL=\"python310.dll\" -DFEAT_MZSCHEME -I 
"C:\racket\include" -DMZ_PRECISE_GC -DDYNAMIC_MZSCHEME -DDYNAMIC_MZSCH_DLL=\"libracket3m_da32r
k.dll\" -DDYNAMIC_MZGC_DLL=\"libracket3m_da32rk.dll\" -DFEAT_PERL -DPERL_IMPLICIT_CONTEXT 
-DPERL_IMPLICIT_SYS -DDYNAMIC_PERL -DDYNAMIC_PERL_DLL=\"perl532.dll\" -DFEAT_RUBY 
-DDYNAMIC_RUBY -DDYNAMIC_RUBY_DLL=\"x64-msvcrt-ruby300.dll\" -DRUBY_VERSION=30 -DFEAT_HUGE 
/Fd.\ObjGXOULYHRZAMD64/ /Zi
Linken: link /nologo /opt:ref /LTCG /HIGHENTROPYVA:NO oldnames.lib kernel32.lib advapi32.lib shell32.lib 
gdi32.lib  comdlg32.lib ole32.lib netapi32.lib uuid.lib user32.lib  /machine:AMD64 version.lib  winspool.lib 
comctl32.lib  libcmt.lib oleaut32.lib  /nodefaultlib:lua54.lib  /STACK:8388608  /nodefaultlib:python27.lib 
/nodefaultlib:python310.lib    winmm.lib WSock32.lib Ws2_32.lib xpm\x64\lib-vc14\libXpm.lib  /PDB:gvim.pdb 
-debug

```

### Minimal vimrc that can reproduce this bug.

Only everforest is needed.

### Steps to reproduce this bug using minimal vimrc

1. Launch vim with "vim -d file1 file2"

### Expected behavior

Being able to see my cursor if I am on whitespace in a line which was added, i.e. green highlighted.

### Actual behavior

My cursor disappears whenever I am on a whitespace character in a line which is highligted as an addition in the diff.

**Comment by sainnhe** [comment]
I can't reproduce, please provide more detailed information (e.g. file1 and file2, the configuration) or screenshots.

**Comment by ibutra** [comment]
Hi,
I investigated further and I must correct myself somewhat:

First 'list' must be set and 'listchars' needs to have something set for spaces, e.g.:
```
set listchars=tab:>-,space:•
```
Then the  cursor does not actually completely disappear, but the contrast is extremely low, so low that I can't see it on some monitors under some lighting conditions.

TL;DR: If the cursor is on a whitespace with 'list' enabled and the whitespace in 'listchars' then the contrast of the cursor is very low

**Comment by sainnhe** [comment]
The information you provided is actually confusing, you said you uses gvim but then you said you executed a command

> Launch vim with "vim -d file1 file2"

Abviously it's from a terminal emulator. So what's exactly vim version are you using? A GUI client like GVim or a TUI client?

This is important because the cursor color in GVim is controlled by a color scheme while the cursor color in a TUI client is controlled by a terminal emulator.

If you are using a TUI client, please try to switch to another terminal color scheme.

**Comment by ibutra** [comment]
I am very sorry for the confusion, I wrote the ticket in a hurry at work. Wont do that again.

I am actually using GVim not Vim. I can  also give you a screenshot now:

![2022-07-26_11h08_47](https://user-images.githubusercontent.com/2822422/181055233-25df03a0-93d4-420f-891d-c301e46425f8.jpg)

Here the cursor is on the right window on line 6 column 3; so two before the printf.
If i set 'nolist' the cursor is nicely bright and yellow. With list on and characters for space the contrast is very low.

**Comment by sainnhe** [comment]
By default, the cursor color is set to reverse, i.e. use the background color in current position as foreground color, and use the foreground color in current position as background color.

https://github.com/sainnhe/everforest/blob/9a8b4f85a7f1bb0e3019911f2c425b994cedb18f/colors/everforest.vim#L78

So what's happening is that the cursor uses the foreground of `listchars` as background which is too dark and hard to see.

To solve this, you can manually set the cursor color to the one you like.

https://github.com/sainnhe/everforest/blob/9a8b4f85a7f1bb0e3019911f2c425b994cedb18f/doc/everforest.txt#L241-L252

**Comment by ibutra** [comment]
Thank you very much. This works nicely!
